crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -23,7 +32,41 @@ pub struct GenericStateUpdate {
     pub reason: Option<String>,
 }
 
+const STATUS: &str = "status";
+const INIT: &str = "init";
+const SUCCESSFUL: &str = "successful";
+const FAILED: &str = "failed";
+const REASON: &str = "reason";
+
 impl GenericCommandState {
+    /// Create an init state for a sub-operation
+    pub fn sub_command_init_state(
+        schema: &MqttSchema,
+        entity: &EntityTopicId,
+        operation: OperationType,
+        cmd_id: CommandId,
+        sub_operation: OperationName,
+    ) -> GenericCommandState {
+        let sub_cmd_id = sub_command_id(&operation.to_string(), &cmd_id);
+        let topic = schema.topic_for(
+            entity,
+            &Channel::Command {
+                operation: OperationType::Custom(sub_operation),
+                cmd_id: sub_cmd_id,
+            },
+        );
+        let status = INIT.to_string();
+        let payload = json!({
+            STATUS: status
+        });
+
```

Comment:
question: What's the difference from just doing this?


```suggestion
        let payload = json!({
            STATUS: status,
            INVOKING_COMMAND: invoking_command_topic.name
        });
```